rust/index/src/spann/types.rs

```diff
@@ -1929,31 +2211,74 @@ impl SpannIndexFlusher {
             max_head_id_id: self.max_head_id_flusher.id(),
             hnsw_id: self.hnsw_id,
         };
+        let num_pl_blocks_flushed = self.pl_flusher.num_blocks();
+        let mut now = std::time::Instant::now();
         self.pl_flusher
             .flush::<u32, &SpannPostingList<'_>>()
             .await
             .map_err(|e| {
                 tracing::error!("Error flushing posting list {}: {}", res.pl_id, e);
                 SpannIndexWriterError::PostingListFlushError(e)
             })?;
+        let mut elapsed = now.elapsed().as_millis() as u64;
+        self.metrics.pl_flush_latency.record(
+            elapsed,
+            &[KeyValue::new(
+                "collection_id",
+                self.collection_id.to_string(),
+            )],
+        );
+        self.metrics.num_pl_blocks_flushed.add(
+            num_pl_blocks_flushed as u64,
+            &[KeyValue::new(
+                "collection_id",
+                self.collection_id.to_string(),
+            )],
+        );
+        let num_versions_map_blocks_flushed = self.versions_map_flusher.num_blocks();
+        now = std::time::Instant::now();
         self.versions_map_flusher
             .flush::<u32, u32>()
             .await
             .map_err(|e| {
                 tracing::error!("Error flushing versions map {}: {}", res.versions_map_id, e);
                 SpannIndexWriterError::VersionsMapFlushError(e)
             })?;
+        elapsed = now.elapsed().as_millis() as u64;
+        self.metrics.versions_map_flush_latency.record(
+            elapsed,
+            &[KeyValue::new(
+                "collection_id",
+                self.collection_id.to_string(),
+            )],
+        );
+        self.metrics.num_versions_map_blocks_flushed.add(
+            num_versions_map_blocks_flushed as u64,
+            &[KeyValue::new(
+                "collection_id",
+                self.collection_id.to_string(),
+            )],
+        );
         self.max_head_id_flusher
             .flush::<&str, u32>()
             .await
             .map_err(|e| {
                 tracing::error!("Error flushing max head id {}: {}", res.max_head_id_id, e);
                 SpannIndexWriterError::MaxHeadIdFlushError(e)
             })?;
+        now = std::time::Instant::now();
```

Comment:
I might ask robert about this, i think he has a cleaner way to do latencies with dropping on a scope instead.